gapic/src/main/com/google/gapid/views/AtomTree.java

```diff
@@ -450,7 +455,13 @@ public void updateElement(Object parent, int index) {
       AtomStream.Node child = ((AtomStream.Node)parent).getChild(index);
       atoms.load(child, refresher::refresh);
       viewer.replace(parent, index, child);
-      viewer.setChildCount(child, child.getChildCount());
+      if (child.getChildCount() > 100) {
+        // For nodes with lots of children, only allocate the children, once the node
+        // is actually expanded.
+        viewer.setHasChildren(child, true);
```

Comment:
It's your call. I'm just allergic to magic constants unless it's immediately obvious what they mean. In this case, I don't know if 100 was a well picked number, especially as I have no idea about the performance differences between the true and false paths.